Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs

```diff
@@ -62,28 +59,49 @@ internal static POMLearner Create(string html, IBrowserElementProvider browserEl
             return new POMLearner(htmlDocument, browserElementProvider, pomSetting, xpathImpl);
         }
 
-        internal Task<IEnumerable<HTMLElementInfo>> LearnElementsAsync()
+        internal Task LearnElementsAsync(IList<ElementInfo> learnedElements)
         {
-            return LearnDocumentElementsAsync(_htmlDocument);
+            return LearnDocumentElementsAsync(_htmlDocument, learnedElements);
         }
 
-        private Task<IEnumerable<HTMLElementInfo>> LearnDocumentElementsAsync(HtmlDocument htmlDocument)
+        private async Task LearnDocumentElementsAsync(HtmlDocument htmlDocument, IList<ElementInfo> learnedElements)
         {
-            return LearnHtmlNodeChildElements(htmlDocument.DocumentNode);
+            await LearnHtmlNodeChildElements(htmlDocument.DocumentNode, htmlNode =>
+            {
+                if (!CheckStaticHtmlNodeExclusion(htmlNode))
+                {
+                    return false;
+                }
+
+                eElementType type = GetElementType(htmlNode);
+                if (_pomSetting != null && !_pomSetting.filteredElementType.Contains(type))
+                {
+                    return false;
+                }
+
+                return true;
+            }, learnedElements);
         }
 
-        private async Task<IEnumerable<HTMLElementInfo>> LearnHtmlNodeChildElements(HtmlNode htmlNode)
+        private async Task LearnHtmlNodeChildElements(HtmlNode htmlNode, Predicate<HtmlNode> filter, IList<ElementInfo> learnedElements)
```

Comment:
Change filter name to more appropriate name

Reply:
handled